actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
```

Comment:
Interesting idea. If I understand you right you want to set hash with all route parts in `defaults_filter_for`.
``` ruby
def defaults_filter_for(pattern)
  {
    controller: /#{options[:pattern]}/
    name: /#{options[:pattern]}/
    verb: /#{options[:pattern]}/
  }
  end
```

But I see one problem. Now, when you call something like this:
```
$ bin/rake routes -c user -g show
```
you'll see only `user_show` route. 

But your solution means that we will have similar hashes for `-c` key and for `-g` key:
``` ruby
# for -c
{ controller: pattern.downcase.sub(/_?controller\z/, '') }

# for -g
{ controller: /#{options[:pattern]}/ }
```

Therefore when we call `bin/rake routes` again, we will see routes only for `user`
controller or routes that match with `show`. What do you think?